### PR TITLE
[tests-only] Group share tests added

### PR DESCRIPTION
**Pull request by Talank**
The following test scenarios have been added:
- Collaborator should not see to whom a file is shared.
- Group sharing
- User (non-author) can not share to a group to which the file is already shared

### Related issues:
- part of https://github.com/owncloud/client/issues/8651

### CODE REVIEW

File: test/gui/shared/steps/steps.py
Comment (swoichha) [inline test/gui/shared/steps/steps.py:677]: We can also mention where this error is to be displayed like you have done in line 466

File: test/gui/shared/steps/steps.py
Comment (swoichha) [inline test/gui/shared/steps/steps.py:685]: This has not been used anywhere. So, you can remove this.

Comment (Talank) [reply test/gui/shared/steps/steps.py:685]: nice find. Thanks

File: test/gui/shared/steps/steps.py
Comment (Talank) [reply test/gui/shared/steps/steps.py:677]: Technically, it looks for an error label that can be found using the selector `{"name": "errorLabel", "type": "QLabel", "visible": 1}`. So, I think it should work for any error in general. 🤔 And IMO mentioning where the error is (for example `... should be displayed in sharing dialog`) brings more confusion.

Comment (swoichha) [reply test/gui/shared/steps/steps.py:677]: If you think this way is better then that's fine. I just mentioned it because this step seems a bit generic.

**Comment by Talank** [comment]
Need to run after scenario to cleanup the state from middleware before merging this PR. So, blocked until https://github.com/owncloud/client/pull/8792 is merged.

**Review by swoichha — APPROVED** [review]
LGTM

**Pull request merged**
